File: server/cxblog-main/src/main/java/me/mcx/modules/blog/domain/Navigation.java
package me.mcx.modules.blog.domain;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import me.mcx.utils.DateUtil;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.*;

import java.io.Serializable;
import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
@TableName("blog_navigation")
@ApiModel(value="Navigation", description="")
public class Navigation implements Serializable {
    private static final long serialVersionUID=1L;


    @ApiModelProperty(value = "主键id")
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;

    @ApiModelProperty(value = "网站分类id")
    private Integer siteClassId;


    @ApiModelProperty(value = "网站标题")
    private String name;


    @ApiModelProperty(value = "网址地址")
    private String url;

    @ApiModelProperty(value = "网站logo")
    private String logo;

    @ApiModelProperty(value = "网站描述")
    private String info;


    @ApiModelProperty(value = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    @JsonFormat(pattern = DateUtil.FORMAT_STRING,timezone="GMT+8")
    private Date createTime;

    @ApiModelProperty(value = "修改时间")
    @TableField(fill = FieldFill.UPDATE)
    @JsonFormat(pattern = DateUtil.FORMAT_STRING,timezone="GMT+8")
    private Date updateTime;
}

File: cxblog-admin/src/api/site/navigation.js
import request from '@/utils/request'

export function getNavigationList(params) {
    return request({
        url: '/api/navigation/list',
        method: 'get',
        params: params
    })
}
export function save(data) {
    return request({
        url: '/api/navigation/insert',
        method: 'post',
        data
    })
}
export function update(data) {
    return request({
        url: '/api/navigation/update',
        method: 'put',
        data
    })
}
export function deleteBatch(data) {
    return request({
        url: '/api/navigation/deleteBatch',
        method: 'delete',
        data
    })
}
